### ISSUE TITLE
Order of features passed to catboost when changed, returns different feature importances (random seed fixed)

### ISSUE THREAD

**Issue by thumbe3** [catboost/catboost#900]
catboost version: 0.15.2
Operating System: Ubuntu

Consider the code run on [titanic dataset's train.csv file](https://www.kaggle.com/c/titanic)
```
import catboost as cb
import pandas as pd
import numpy as np

df = pd.read_csv('train.csv')
df = df.fillna(999)

X_train = df.drop('Age', axis=1)
#commenting and uncommenting this line yields different results
#X_train = X_train.reindex(sorted(X_train.columns), axis=1)
y_train =df['Age']

cat_feature_types = [X_train[col].dtype != np.float for col in X_train.columns]
cat_feature_indices = [i for i, x in enumerate(cat_feature_types) if x]
train_pool = cb.Pool(data=X_train, label=y_train, cat_features=cat_feature_indices)
params = {'max_depth': 10, 'eta': 0.5, 'silent': False, 'objective': 'MultiClass',
                      'random_seed': 42, 'od_type': 'Iter', 'od_wait': 20, 'boosting_type': 'Plain',
                      'bootstrap_type': 'Bernoulli', 'subsample': 0.66, 'max_ctr_complexity': 2, 'iterations':1000}
cbmodel = cb.CatBoostClassifier(**params)
cb_model = cbmodel.fit(train_pool, verbose=True)
feature_importances = cbmodel.get_feature_importance()
print(feature_importances)
```

The commented line of code if uncommented returns different feature_importances which is basically sorting the columns according to column names. I have set the random_seed value as well. Is this behavior desirable?

**Comment by annaveronika** [comment]
Order of features should have same effect as changing random seed, because it influences order of random number generations. Could you check that?

**Comment by thumbe3** [comment]
Thanks for the swift response. I checked that, and it indeed is the case.
 But, something I don't get is how does the order of feature importances change just by changing the random seed. I see that,sometimes,  the top 3 features are different with different random seed as the argument.

**Comment by annaveronika** [comment]
Changes in the order are fine, if the difference in the value is not big. But even if it is big, it happens sometimes. Also you could try other feature importances. The default ones show how much predictions change, if feature value changes. There are also ones that show how much loss would change if you would not have this feature: https://catboost.ai/docs/concepts/fstr.html#fstr__regular-feature-importances__lossfunctionchange

**Comment by thumbe3** [comment]
Thanks for the link!
